lib/datadog/tracing/span_operation.rb

```diff
@@ -439,6 +443,11 @@ def message
       # mutation by reference; when this span is returned,
       # we don't want this SpanOperation to modify it further.
       def build_span
+        # Use parent service if no service is set and @inherit_parent_service is set
```

Comment:
I am generally supportive for moving toward this direction. What I found a larger problem is the fact that `service` is a mutable field. 

Things get tricky when parent span changes its service after child spans are finished. For instance:

```ruby

Datadog::Tracing.trace('parent', service: 'service-parent') do |span_op, _|
  Datadog::Tracing.trace('child1') {}
  span_op.service = "BalaBoom!" # Change the service name after the child span is finished
end
```

I was not sure what the behaviour is like for instrumentation depending on `ActiveSupport::Notifications`, which I remember the service field is set in the after callback. 

Any thoughts about restricting the mutability of the service field?

Reply:
I agree with Tony; mutability is a problem. Generally speaking, I think moving towards an immutable world simplifies things quite a bit, the only question is how.

Reply:
Secondary question: how does this feature fit in with the "one service name" change we originally planned for 2.0? If we end up introducing that, is this obsolete?